### PR TITLE
Added option description setter

### PR DESCRIPTION
**Pull request by helmesjo**
Option description can be modified after adding option.

Related: https://github.com/CLIUtils/CLI11/issues/193

### CODE REVIEW

File: include/CLI/Option.hpp [hunk 1]
```diff
@@ -517,6 +517,9 @@ class Option : public OptionBase<Option> {
     /// Get the description
     const std::string &get_description() const { return description_; }
 
+    /// Set the description
+    void description(const std::string& description) { description_ = description; }
```
Comment (henryiii) [inline include/CLI/Option.hpp]: Could you please return Option* (`this`)? That way these can be chained.

Comment (henryiii) [reply include/CLI/Option.hpp]: Don't worry about style, but if you are editing it anyway, `std::string &description` will make clang-format happy.

**Review by henryiii — COMMENTED** [review]
One minor comment. It would be nice to be able to edit the app description too.

File: include/CLI/Option.hpp (hunk 1 above)
Comment (helmesjo) [reply include/CLI/Option.hpp]: Np, I'll fix!

**Comment by henryiii** [comment]
If that passes CI, I'll merge. Are you planning to make one for App description, too?

**Comment by helmesjo** [comment]
> Are you planning to make one for App description, too?

Done!

**Comment by henryiii** [comment]
That looks great, thank you!

**Comment by helmesjo** [comment]
What's the ETA on the next release?

**Comment by henryiii** [comment]
Soon, 1 day to 1 week. Basically consider the current master an RC. Depends on how much testing I can do or if bugs are found.

**Comment by helmesjo** [comment]
Great!

**Comment by helmesjo** [comment]
When it's released, could you create an issue in [bincrafters-community](https://github.com/bincrafters/community/issues) notifying about this? The conan package in question is [conan-cli11](https://github.com/bincrafters/conan-cli11). Or, just ping me and I'll do it. :)

**Comment by henryiii** [comment]
I'm going to see if I can get #200 in. It's a name change close to release, but the old behavior was dangerous. I'll ping you or open an issue when 1.7 is out! (PS: new releases show up in RSS form here: 
<https://github.com/CLIUtils/CLI11/releases.atom>, or GitHub now lets you `Watch -> Releases Only` (which I have not tried, because I use the RSS feeds to watch projects, but sounds useful).

**Comment by helmesjo** [comment]
Sweet! 
Oh yeah, forgot about that feature! 😇

**Comment by henryiii** [comment]
It's out! Can you raise the issue?

**Comment by helmesjo** [comment]
I'll fix!

**Comment by henryiii** [comment]
Just curious, how is bincrafters different from conan? I already have an official conan package that gets pushed out by the CI on tags.

**Comment by helmesjo** [comment]
Oh, alright. Yeah I can see you [build conan packages](https://github.com/CLIUtils/CLI11/blob/3cb75b903e12b31acd26075110c5bba64f9d04f8/.appveyor.yml#L19), but they are never pushed to any repository, right? 

Nah [bincrafters](https://github.com/bincrafters/ ) is just a group of people doing great work of writing/collecting conan packages/recipes, build them on travis/appveyor and push all of the pre-built packages to [their own repository](https://bintray.com/bincrafters/public-conan) on bintray.

Bincrafters host sort of an inofficial "official" conan repository, and over time packages are included in the official [conan repository](https://bintray.com/conan/conan-center).

Perhaps @Croydon have some thoughts about where to go from here?

**Comment by henryiii** [comment]
Not sure how repositories work. I'm not using Conan at the moment, just providing packages. I'm happy to have them in bincrafters, just was checking to make sure work wasn't duplicated if unnecessary.

Packages are here: https://bintray.com/cliutils/CLI11

**Comment by helmesjo** [comment]
Ah, nice! I'll hand it over to @Croydon to take this forward. :) I guess you could submit an inclusion request into bincrafters, with you as official maintainer. That recipe would then solely wrap [yours](https://github.com/CLIUtils/CLI11/blob/master/conanfile.py), or something like that. I don't know the specifics but something similar is being done for [magnum](https://github.com/mosra/magnum/issues/304#issuecomment-451692423).

**Comment by helmesjo** [comment]
In the meantime: https://github.com/bincrafters/community/issues/619

**Comment by Croydon** [comment]
Hi! :) 

I didn't follow this project so far, but from a quick look it seems like everything is already fine. Since CLI11 is header-only packaging + deploying is pretty simple. The recipe could need some more meta data like topics, author, homepage and the license field should be a SPDX identifier (https://spdx.org/licenses/), in general I recommend to have a look at https://github.com/bincrafters/conan-templates/blob/2dbf5261858c817faea420b3c32b7436f204fd85/header_only/conanfile.py#L8

CLI11 is already in conan-center so I'm not seeing any major things left to do here. @helmesjo What did you had in mind? Maybe the confusion comes because at Bincrafters we had a recipe as well, but we are stopping maintaining those when there is official support https://github.com/bincrafters/conan-cli11

**Comment by helmesjo** [comment]
Oh, I totally missed that it is already in conan-center..! :man_facepalming: I've been using the bincrafters-package since way back so just figured that was still the one... :) 

Sorry for the confusion people! :blush:

**Comment by henryiii** [comment]
@Croydon Thanks, I'll update the metadata soon (traveling for a few days).

**Comment by helmesjo** [comment]
So, I'm a little confused... (this is a conan/bintray question, but related to this release):

@Croydon mentioned that it is already part of conan-center, which one can see [here](https://bintray.com/conan/conan-center) when searching for `CLI11`. It is a so called "linked package" which is [described by bintray](https://www.jfrog.com/confluence/display/BT/Promoting+Your+Material) as:
> Linking a package to another user’s repository. A package that belongs to one repository may be linked and displayed as if it also belongs to another repository. This means that your package will be downloaded, visible in search results, and promoted just as much as the repository to which it is linked (in addition to, not instead of, the repository to which it actually belongs). This is one more option Bintray provides for cooperation between developers.

Given this, why can't I see this package locally when searching for `cli11*` on remote `conan-center`?
All I get is:
```
cli11/1.3.0@bincrafters/stable
cli11/1.4.0@bincrafters/stable
cli11/1.5.3@bincrafters/stable
cli11/1.6.0@bincrafters/stable
cli11/1.6.1@bincrafters/stable
```

Remote: `https://api.bintray.com/conan/conan/conan-center` (using the url displayed on bintray as `https://dl.bintray.com/conan/conan-center` does not work, it gives error `ERROR: Not implemented endpoint. [Remote: https://conan.bintray.com]`).

I'm missing some piece of the puzzle...

**Comment by henryiii** [comment]
Is there a way to update your cache, perhaps? I know Conan center/bin tray are very picky about caps, but that should be fine for the last several releases. I’m traveling without my computer, can test in a few days.

**Comment by Croydon** [comment]
> conan search CLI11* -r conan-center

works for me as expected. The search is case-sensitive as such

> conan search cli11* -r conan-center

only returns the results of our Bincrafters' recipe

**Comment by helmesjo** [comment]
Oh, wow. Case sensitivity totally got me... :man_facepalming: Yep, now I see it! :)

Thanks!

**Pull request merged**